File: P1/ejercicio4/PasswordUpperCaseFilter.java
package ejercicio4;

public class PasswordUpperCaseFilter implements Filter {
    @Override
    public void execute(Message message) throws Exception {
        String password = message.getPassword();
        if (password == null || !password.matches(".*[A-Z].*")) {
            throw new Exception("Contraseña inválida: debe contener al menos una mayúscula");
        }
    }
}

File: P2/email_filter_app/lib/filters/password_upper_case_filter.dart
import 'package:email_filter_app/credentials.dart';
import 'package:email_filter_app/filters/filter.dart';

class PasswordUpperCaseFilter extends Filter {
  PasswordUpperCaseFilter() : super("Validar mayúscula en contraseña");

  @override
  void execute(Credentials credentials) {
    final password = credentials.password;
    if (password.isEmpty || !RegExp(r'[A-Z]').hasMatch(password)) {
      throw Exception(
        'Contraseña inválida: debe contener al menos una mayúscula',
      );
    }
  }
}
